File: problems/algorithms/array-prime-numbers/ruby/response/src/prime_numbers.rb
require_relative "../../../../function-prime/ruby/response/src/function_prime"

=begin
  begin: 1..n
  end: 1..n, end > begin
=end

def primes(bg:nil, ed:)
  result = []
  if bg != nil 
    if (ed < bg)
      return result
    end
    (bg..ed).each do |num|
      result << num if isPrime(num)
    end
  else
    i = 2
    while result.length < ed
      if isPrime(i)
        result.push(i)
      end
      i += 1
    end
  end
  return result
end

File: src/pages/exercises/array-prime-numbers/codes/js/response/prime-numbers.js
import { isPrime } from '../../../function/prime/response/prime.js';

/**
 * begin: 1..n
 * end: 1..n, end > begin
 */
function primes(begin, end) {
  let result = [];

  if (end < begin) return result;

  if (end) {
    for (let i = begin; i <= end; i++) if (isPrime(i)) result.push(i);
  } else {
    for (let i = 2; result.length < begin; i++) if (isPrime(i)) result.push(i);
  }

  return result;
}

export { primes };
